### PR TITLE
[rush] Part 1 of "litewatch" plugin prototype

### PR DESCRIPTION
**Pull request by octogonz**
## Summary

Litewatch is an experimental Rush plugin based loosely on the idea behind @dimfeld's https://github.com/dimfeld/rush-dev-watcher tool.  The main difference is that my implementation uses a more sophisticated algorithm for detecting and suppressing irrelevant output (for example, a downstream project that is failing to build because of upstream failures).

## Details

To facilitate code review, I'm splitting my work into multiple PRs.  This PR just sets up the project and commits the scheduling engine.

### CODE REVIEW

File: rush-plugins/rush-litewatch-plugin/config/rush-project.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "operationSettings": [
```
Comment (iclanton) [inline rush-plugins/rush-litewatch-plugin/config/rush-project.json:2]: Why not just use `heft-node-rig`?

Comment (iclanton) [reply rush-plugins/rush-litewatch-plugin/config/rush-project.json:2]: Ah you are. This file shouldn't be necessary.

Comment (octogonz) [reply rush-plugins/rush-litewatch-plugin/config/rush-project.json:2]: This is a copy+paste of the same `rush-project.json` used by all the other projects in this repo.  Should we delete it for all projects?  (Maybe in a separate PR?)

Comment (octogonz) [reply rush-plugins/rush-litewatch-plugin/config/rush-project.json]: Maybe those projects have cyclic dependencies.  I removed this file and there weren't any build cache warnings.

**Review by iclanton — APPROVED** [review]

File: rush-plugins/rush-litewatch-plugin/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  "private": true,
+  "description": "An experimental alternative approach for multi-project watch mode",
+  "license": "MIT",
+  "repository": {
+    "url": "https://github.com/microsoft/rushstack.git",
+    "type": "git",
+    "directory": "rush-plugins/rush-litewatch-plugin"
+  },
+  "scripts": {
+    "build": "heft test --clean"
```
Comment (iclanton) [inline rush-plugins/rush-litewatch-plugin/package.json:13]:
```suggestion
    "build": "heft test --clean",
    "_phase:build": "heft build --clean",
    "_phase:test": "heft test --no-build"
```

File: rush-plugins/rush-litewatch-plugin/src/WatchManager.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { WatchProject, WatchState } from './WatchProject';
+import { ITerminal, ITerminalProvider, Terminal } from '@rushstack/node-core-library';
```
Comment (iclanton) [inline rush-plugins/rush-litewatch-plugin/src/WatchManager.ts:5]:
```suggestion
import { ITerminal, ITerminalProvider, Terminal } from '@rushstack/node-core-library';

import { WatchProject, WatchState } from './WatchProject';
```

**Review by iclanton — APPROVED** [review]

**Pull request merged**